src/screens/editMonitoring/hooks/useEditMonitoring.ts

```diff
@@ -0,0 +1,287 @@
+import { useSnackBar } from '../../../utils/renderSnackBar';
+import { useState, useMemo, useEffect } from 'react';
+import { TWeekDayAvailabilityState } from './types';
+import {
+  WEEK_DAY_AVAILABILITY_INITIAL_STATE,
+  END_HOURS_OPTIONS,
+  START_HOURS_OPTIONS,
+} from './contants';
+import { TAvailability } from '../../../service/requests/useUpdateMonitorAvailabilityRequest/types';
+import useGetMonitorRequest from '../../../service/requests/useGetMonitorRequest';
+import { SelectChangeEvent } from '@mui/material';
+import usePatchMonitorRequest from '../../../service/requests/usePatchMonitorRequest';
+import useGetLoggedUser from '../../../service/storage/getLoggedUser';
+import useQuery from '../../../utils/useQuery';
+import { useNavigate } from 'react-router-dom';
+import { SCREENS } from '../../../utils/screens';
+
+const useEditMonitoring = () => {
+  const navigate = useNavigate();
+  const query = useQuery();
+  const user = useGetLoggedUser();
+  const { showSuccessSnackBar, showErrorSnackBar } = useSnackBar();
+
+  const {
+    getMonitor,
+    data: monitor,
+    isLoading: isLoadingGetMonitor,
+  } = useGetMonitorRequest();
+  const {
+    isLoading: isLoadingPatch,
+    isSuccess: isSuccessPatch,
+    error: errorPatch,
+    patchMonitor,
+  } = usePatchMonitorRequest();
+
+  const [preference, setPreference] = useState('');
+  const [editMode, setEditMode] = useState(
+    query.get('edit') === 'true' ? true : false
+  );
+  const [isOpenCancelModal, setIsOpenCancelModal] = useState(false);
+  const [weekDayAvailability, setWeekDayAvailability] = useState<
+    TWeekDayAvailabilityState[]
+  >(WEEK_DAY_AVAILABILITY_INITIAL_STATE);
+
+  const resetWeekDay = () => {
+    setWeekDayAvailability((prev) =>
+      prev.map((weekDay) => ({
+        ...weekDay,
+        isSelected: false,
+        fromHourIndex: -1,
+        toHourIndex: -1,
+      }))
+    );
+  };
+
+  const setupOldAvailability = () => {
+    if (monitor) {
+      const oldAvailability: TWeekDayAvailabilityState[] =
+        WEEK_DAY_AVAILABILITY_INITIAL_STATE;
+
+      monitor.availability.map((weekDay) => {
+        oldAvailability[weekDay.week_day] = {
+          weekDay: weekDay.week_day,
+          name: oldAvailability[weekDay.week_day].name,
+          label: oldAvailability[weekDay.week_day].label,
+          isSelected: true,
+          fromHourIndex: START_HOURS_OPTIONS.indexOf(weekDay.start),
+          toHourIndex: END_HOURS_OPTIONS.indexOf(weekDay.end),
+        };
+      });
+
+      setWeekDayAvailability(oldAvailability);
```

Comment:
Quando a tela é aberta com `edit=true` e você não altera os horário de disponibilidade, ao tentar salvar a rota dá erro porque é enviado um array de disponibilidades vazio.

Isso acontece porque quando você carrega os dados da rota e atualiza o `weekDayAvailability` aqui, você tá passando o valor pro estado, que é o endereço da lista `WEEK_DAY_AVAILABILITY_INITIAL_STATE`. Sendo assim, o estado não é alterado e o `availabilities` enviado na rota também não é atualizado com as disponibilidades, já que o estado que o useMemo está observando não foi alterado.

Você precisa ter bastante cuidado com estados de tipo complexo por causa disso, alterar um índice de um array ou a chave de um objeto não altera o estado, porque o endereço do array/objeto não mudou. Esse é um dos motivos de eu ter falado em outro comentário sobre como atualizar o valor de um estado quando o novo valor depende do anterior.

Para resolver esse problema específico, só de fazer a desconstrução da lista já vai criar uma outra lista com endereço diferente, aí o estado vai ser atualizado e tudo vai dar certo:

```suggestion
      setWeekDayAvailability([...oldAvailability]);
```